trino/trino.go

```diff
@@ -209,7 +209,125 @@ type Config struct {
 	QueryTimeout               *time.Duration    // Configurable timeout for query (optional)
 }
 
-// FormatDSN returns a DSN string from the configuration.
+func ParseDSNToConfig(dsn string) (*Config, error) {
+	serverURL, err := url.Parse(dsn)
```

Comment:
Does this duplicate `newConn(dsn string)`? Maybe we can call it there and simplify setting up the connection by taking values from the Config struct.